File: scripts/js/list_files_in_container.js
const { BlobServiceClient } = require('@azure/storage-blob');

async function listFilesInContainer(sasToken, containerName) {
    try {
        const blobServiceClient = new BlobServiceClient(`https://msgedcstorage.blob.core.windows.net?${sasToken}`);
        const containerClient = blobServiceClient.getContainerClient(containerName);

        console.log(`Files in container '${containerName}':`);
        for await (const blob of containerClient.listBlobsFlat()) {
            console.log(blob.name);
        }
    } catch (error) {
        console.error(`Error listing files in the container: ${error.message}`);
    }
}

if (process.argv.length !== 3) {
    console.log('Usage: node list_files_in_container.js <container_name>');
    process.exit(100);
}

const containerName = process.argv[2];
const sasToken = process.env.SAS_TOKEN;

if (!sasToken) {
    console.error('SAS token not found in environment variable SAS_TOKEN.');
    process.exit(101);
}

listFilesInContainer(sasToken, containerName);

File: scripts/python/list_files_in_container.py
from azure.storage.blob import BlobServiceClient
import os
import sys


# Function to list all files in the Blob storage container
def list_files_in_container(sas_token, container_name):
    try:
        # Create a Blob Service Client
        blob_service_client = BlobServiceClient(account_url="https://msgedcstorage.blob.core.windows.net",
                                                credential=sas_token)

        # Create a Container Client
        container_client = blob_service_client.get_container_client(container_name)

        # List all blobs in the container
        blobs_list = container_client.list_blobs()
        print(f"Files in container '{container_name}':")
        for blob in blobs_list:
            print(blob.name)

    except Exception as e:
        print(f"Error listing files in the container: {e}")


if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python list_files.py <container_name>")
        sys.exit(100)

    # Container name from command line argument
    container_name = sys.argv[1]

    # Retrieve SAS token from environment variable
    sas_token = os.getenv('SAS_TOKEN')
    if not sas_token:
        print("SAS token not found in environment variable SAS_TOKEN.")
        sys.exit(101)

    # List files in the container
    list_files_in_container(sas_token, container_name)
